src/tablestore.rs

```diff
@@ -98,11 +110,24 @@ impl TableStore {
             filter_cache: RwLock::new(HashMap::new()),
             transactional_wal_store: Arc::new(DelegatingTransactionalObjectStore::new(
                 Path::from("/"),
-                object_store.clone(),
+                cacheable_object_store.as_direct(),
             )),
         }
     }
 
+    pub(crate) fn as_uncached(&self) -> Arc<TableStore> {
+        Arc::new(TableStore {
```

Comment:
Hmm, not sure how I feel about creating new table stores every time. The filter cache is reset on each call. Seems like it'd be better to keep that. Even when we move it out, doing the new on each `as_uncached` means tablestore has to stay stateless. 🤔